osbs/utils.py

```diff
@@ -327,20 +328,34 @@ def make_name_from_git(repo, branch, limit=53, separator='-', hash_size=5):
 
     size = (2 * len(separator)) + hash_size
 
+    disallowed = '._'
     for i in range(max(len(repo), len(branch))):
         for group, group_chars in groups:
             if i < len(group):
-                group_chars.append(group[i])
+                char = group[i]
+                if char in disallowed:
+                    # Skip disallowed chars altogether
+                    continue
+
+                group_chars.append(char)
                 size += 1
                 if size >= limit:
                     break
         else:
             continue
         break
 
+    components = []
     repo = ''.join(repo_chars)
+    if repo:
```

Comment:
why the if? there would be always at least 1 valid character, I don't think that git would allow you name without a-z0-9 char